### DISCUSSION TITLE
Imposing a BC that includes a time derivative of the main variable

### DISCUSSION THREAD

**Discussion by alimostafavi24** [idaholab/moose#26657] [Q&A General]
Dear MOOSE community,

Is it possible to have a boundary condition that accounts for the time derivative of the main variable? I want to impose the following boundary condition (third equation in the following picture and the main variable is also phi). In the 2D problem, this condition would be on the bottom boundary which is a horizontal line. Is there any similar condition related to this specific problem in the framework?

![Screenshot from 2024-01-27 11-35-30](https://github.com/idaholab/moose/assets/119982603/c42d3b6e-208d-49a9-b08d-044a3dc53012)

Thanks,
Ali

**Comment by GiudGiud** [comment]
Hello

Yes this should be doable. Use the routines from the coupleable interface (coupledDot) to get the contribution to the residual right. 

I think to get it completely right you ll want to also contribute to the "time" matrix and vectors. See the TimeKernel.C for how this is done (changing the tags). 

Guillaume
